### PR TITLE
Make generateSecret more robust

### PR DESCRIPTION
**Pull request by morgen-peschke**
1. Handle non-200 successes
2. Attempt recovery if the body is malformed/omitted

Addresses #392

### CODE REVIEW

File: core/src/main/scala/com/banno/vault/Vault.scala [hunk 1]
```diff
@@ -308,19 +308,38 @@ object Vault {
...
+                  extra = s"tokenLength=${token.length}".some
+                )
+              }
+          }
+      case resp =>
         F.raiseError(
           VaultRequestError(
             request = withBody,
-            cause = e.some,
+            cause =
```
Comment (bannopeter) [inline core/src/main/scala/com/banno/vault/Vault.scala:337]: are we losing any clues/reasons that may have been in the response body? (I have no idea, just wondering)

Comment (morgen-peschke) [reply core/src/main/scala/com/banno/vault/Vault.scala:337]: I don't _think_ so. I traced the previous `Client[F]#expect` call back to `org.http4s.client.DefaultClient#defaultOnError`, which appears to discard the body.

File: core/src/test/scala/com/banno/vault/VaultSpec.scala [hunk 2]
```diff
@@ -627,6 +668,92 @@ class VaultSpec
...
+              leaseId.some,
+              renewable.some
+            )
+          )
+    }
+  }
+
+  test(
+    "generateSecret works as expected when receiving a 200 Ok with an unparsable response"
+  ) {
```
Comment (rossabaker) [inline core/src/test/scala/com/banno/vault/VaultSpec.scala:693]: I'm lost what made that test unparsable.

Comment (morgen-peschke) [reply core/src/test/scala/com/banno/vault/VaultSpec.scala:693]: The third type argument to `generateSecret` on line 697 doesn't match the second, so it's putting in a `SelfManaged` and trying to pull it back out as a `VaultKeys` (which has an incompatible serialization).

**Pull request merged**